src/evict/evict_lru.c

```diff
@@ -594,12 +594,8 @@ __evict_pass(WT_SESSION_IMPL *session)
 		 * If there is a request to clear eviction walks, do that now,
 		 * before checking if the cache is full.
 		 */
-		if (F_ISSET(cache, WT_CACHE_CLEAR_WALKS)) {
-			F_CLR(cache, WT_CACHE_CLEAR_WALKS);
-			WT_RET(__evict_clear_walks(session));
-			WT_RET(__wt_cond_signal(
-			    session, cache->evict_waiter_cond));
-		}
+		if (F_ISSET(cache, WT_CACHE_CLEAR_WALKS))
```

Comment:
It is used now to cause the code to know a thread wants to clear its walks so the thread doing the populate can stop as soon as possible.  Otherwise a thread may way for an entire populate and evict pass to complete.  That seems like it could cause stalls (although I haven't measured that).